anomalib/config/config.py

```diff
@@ -140,6 +141,19 @@ def get_configurable_parameters(
 
     # Project Configs
     project_path = Path(config.project.path) / config.model.name / config.dataset.name
+
+    if "unique_dir" not in config.project.keys():
+        warn("config.project.unique_dir not found config file. Setting to False for backward compatibility")
+        config.project.unique_dir = False  # backward compatibility
+
+    if config.project.unique_dir:
+        project_path = project_path / datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+    else:
+        warn(
+            "config.project.unique_dir is set to False. "
+            "This does not ensure that your results will be written in an empty directory and you may overwrite files."
+        )
+
```

Comment:
It would be better to place this after line 158, so that the category folder comes before the unique folder.

Reply:
Ah yess. Thanks.

Btw, another idea : would it make sense to name it (in addition to the datetime) a hash of the config file? 

Like this you know that two folders were run with exactly the same options.

Reply:
As long as it doesn't lead to excessively long and unreadable folder names I'm fine with adding it. We could also store a copy of the config in the folder to make it easier to find back a specific run.